paasta_tools/tron_tools.py

```diff
@@ -1169,15 +1170,59 @@ def validate_complete_config(
     return []
 
 
-def get_tron_namespaces(cluster, soa_dir):
+class ActionInstanceConfigDict(InstanceConfigDict):
+    executor: str
+    command: str
+
+
+def _is_action_spark(action_info: ActionInstanceConfigDict) -> bool:
+    if action_info.get("executor", "") == "ssh" and "spark-submit" in action_info.get(
+        "command", ""
+    ):
+        return True
+    return False
+
+
+def _is_action_regular_tron(action_info: ActionInstanceConfigDict) -> bool:
+    return action_info.get("executor", "") in ["paasta", ""]
+
+
+def get_tron_namespaces(
+    cluster: str,
+    soa_dir: str,
+    tron_executors: List[str] = ["paasta"],
+) -> List[str]:
     tron_config_file = f"tron-{cluster}.yaml"
     config_dirs = [
         _dir[0]
         for _dir in os.walk(os.path.abspath(soa_dir))
         if tron_config_file in _dir[2]
     ]
     namespaces = [os.path.split(config_dir)[1] for config_dir in config_dirs]
-    return namespaces
+    spark_namespaces = set()
+    for service in namespaces:
+        config = filter_templates_from_config(
+            read_extra_service_information(
+                service,
+                extra_info=f"tron-{cluster}",
+                soa_dir=soa_dir,
+                deepcopy=False,
+            )
+        )
+        for job_name, job in config.items():
+            for action, action_info in job.get("actions", {}).items():
+                if "spark" in tron_executors and _is_action_spark(action_info):
+                    spark_namespaces.add(service)
+                    break
+                elif "paasta" in tron_executors and _is_action_regular_tron(
+                    action_info
+                ):
+                    spark_namespaces.add(service)
+                    break
+            # No need to iterate other jobs if at least one job or action qualified the service
+            if service in spark_namespaces:
+                break
```

Comment:
it might be a little cleaner to do something like iterate over all configs and create a dict of `{executor_type: [list/set, of, services, ...]}` and then just concatenate the requested ones for the return value

e.g.,
```
services_by_executor_type = defaultdict(set)
for service in namespaces:
        config = filter_templates_from_config(
            read_extra_service_information(
                service,
                extra_info=f"tron-{cluster}",
                soa_dir=soa_dir,
                deepcopy=False,
            )
        )

        for job_name, job in config.items():
            for action, action_info in job.get("actions", {}).items():
               
 services_by_executor_type[get_executor_type(action_info].add(service)

retval = set()
for executor_type in executor_types:
    retval |= services_by_executor_type[executor_type]
return retval
```

or something like that :p